crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -108,6 +108,15 @@ impl<T: TransactionOrdering> TxPool<T> {
         self.all().txs_iter(sender).last().map(|(_, tx)| Arc::clone(&tx.transaction))
     }
 
+    /// Returns the next valid transaction for the given sender.
+    pub(crate) fn get_next_valid_transaction_by_sender(
+        &self,
+        sender: SenderId,
+        on_chain_nonce: u64,
+    ) -> Option<Arc<ValidPoolTransaction<T::Transaction>>> {
+        self.all().txs_iter(sender).filter(|(_, tx)| tx.transaction.nonce() > on_chain_nonce).map(|(_, tx)| Arc::clone(&tx.transaction)).next()
```

Comment:
this is slightly more complex because we need to find the last tx that is connected to on_chain_nonce by +1, only checking `>` is not correct.

this can be solved best with a for loop

Reply:
Do the function signature looks good to you? I'm quite confusing on how to name this function :D

Reply:
probably the updated one is not correct as your description